core/src/main/java/com/linecorp/armeria/common/stream/FilteredStreamMessage.java

```diff
@@ -198,7 +201,12 @@ public void onNext(T o) {
 
         @Override
         public void onError(Throwable t) {
+            // Should call beforeError() before checking CancelledSubscriptionException so that
+            // the subclass can clean up the resources.
             final Throwable filteredCause = beforeError(delegate, t);
+            if (t instanceof CancelledSubscriptionException && !notifyCancellation) {
+                return;
+            }
```

Comment:
From the Javadoc of `beforeError()`, don't we need to call `Subscriber.onError()` if `beforeError()` is called. 🤔
> * A callback executed just before calling {@link Subscriber#onError(Throwable)} on {@code subscriber}.

Reply:
I also added `This callback is also executed when {@link Subscription#cancel()}...`.
Let me rephrase it a bit. 😄